paramiko/channel.py

```diff
@@ -989,7 +989,7 @@ def _handle_request(self, m):
             else:
                 ok = server.check_channel_env_request(self, name, value)
         elif key == 'exec':
-            cmd = m.get_text()
+            cmd = m.get_string()
             if server is None:
                 ok = False
             else:
```

Comment:
type(cmd) changes from unicode to bytes. If strict compatibility with existing code is a concern, the following could be a solution:
```
    from paramiko.py3compat import u
    try:
        cmd = u(cmd)
    except UnicodeDecodeError:
        pass
```

This way the behaviour does not change, if the ssh-client sends valid UTF-8. If cmd is not UTF-8, it will be passed on as bytes. A server, that wants command to be bytes, could use the following ServerInterface.check_channel_exec_request implementation:
```
    def check_channel_exec_request(self, channel, command):
        if isinstance(command, unicode):
            command = command.encode('utf-8')
        ...
```